### ISSUE TITLE
Enhancement: file.tidy to remove unwanted files from a directory tree

### ISSUE THREAD

**Issue by dhs-rec** [saltstack/salt#46752]
I'm in a situation where I need to remove some unwanted files from a directory tree, without knowing at which level they are stored (could also be multiple files matching a glob pattern at different levels in the tree). So it would be nice to have something similar to [Puppet's tidy resource](https://puppet.com/docs/puppet/4.10/type.html#tidy).

For example:
```
remove_foo:
  file.tidy:
    - name: /path/to/directory
    - recurse: True
    - matches: foo
```
This would remove all files named foo at any level below /path/to/directory.

**Comment by dhs-rec** [comment]
@gtmanfred, thanks for approving.

**Comment by cachedout** [comment]
@gtmanfred Should this really be a separate state function or would it be better simply to roll this functionality into `file.absent`?

**Comment by gtmanfred** [comment]
Makes sense it could be rolled into file.absent.

**Comment by dhs-rec** [comment]
@gtmanfred, @cachedout: I really think it shouldn't because file.absent removes the given file/directory, whereas file.tidied removes items *FROM* the given directory tree, which is a completely different thing.

**Comment by gtmanfred** [comment]
How is that different from just saying, only mark things as absent recursively from a directory that match these regexes?

**Comment by dhs-rec** [comment]
Guess this has meanwhile been released, right?

**Comment by dhs-rec** [comment]
To answer my own question: Yes, it has, as of 2019.2.0. Anyway, is there any chance to have it backported to 2018.3.x?

**Comment by gtmanfred** [comment]
based on changes to the branching schemas here and what is supported going forward, probably not https://github.com/saltstack/salt-enhancement-proposals/pull/20

**Comment by dhs-rec** [comment]
This is already released with 2019.2.x, closing.
